libs/common/result.hpp

```diff
@@ -106,8 +107,73 @@ namespace iroha {
                               std::forward<ValueMatch>(value_func),
                               std::forward<ErrorMatch>(error_func));
       }
+
+      constexpr bool has_val() const {
+        return some_val();
+      }
+
+      constexpr bool has_err() const {
+        return some_err();
+      }
+
+      constexpr auto some_val() const {
+        return visit_in_place(
+            *this,
+            [](ValueType v) { return boost::optional<ValueType>(v); },
+            [](ErrorType e) -> boost::optional<ValueType> { return {}; });
+      }
+
+      constexpr auto some_err() const {
+        return visit_in_place(
+            *this,
+            [](ValueType v) -> boost::optional<ErrorType> { return {}; },
+            [](ErrorType e) { return boost::optional<ErrorType>(e); });
+      }
+
+      /**
+       * @return new_res if this Result contains a value
+       *         otherwise return this
+       */
+      template <typename Value>
+      constexpr Result<Value, E> and_res(Result<Value, E> new_res) const {
```

Comment:
I'm not sure this is best obvious name, do you have any other candidates?

Reply:
Not really, in rust it named just `and`, but that is a keyword is c++